### PR TITLE
Add new flow for deleting provider users

### PR DESCRIPTION
**Pull request by benswannack**
## Context
With the new scoped-to-provider users page, we need a new flow for removing provider users

## Changes proposed in this pull request
- Add new routes for deleting provider users
- Add new service for removing a user from a specified org

## Guidance to review
Per commit, test in the review app

## Link to Trello card
https://trello.com/c/IMbZ8zrQ/4091-delete-provider-user

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Review by SamJamCul — COMMENTED** [review]
Looks good to me - I'm going to wait until the PR underneath is merged in

File: app/views/provider_interface/users/show.html.erb
Comment (cwrw) [inline app/views/provider_interface/users/show.html.erb:20]: maybe we could move this to the locales?

**Review by cwrw — APPROVED** [review]
Very cleanly done, looks great! I was wondering about users deleting themselves from organisations, if you end up doing that you get into a very weird place. Should we allow users to delete themselves in the first place, if so can we improve the journey? something to discuss with the design team (and can be in a subsequent ticket) as its still part of the current behaviour

File: app/views/provider_interface/users/show.html.erb
Comment (benswannack) [reply app/views/provider_interface/users/show.html.erb]: 👍 done the same for the equivalent org permissions one

**Pull request merged**
